frappe/public/js/frappe/form/save.js

```diff
@@ -274,6 +274,23 @@ frappe.ui.form.save = function (frm, action, callback, btn) {
 	}
 };
 
+frappe.ui.form.discard = function (frm, callback, btn) {
```

Comment:
Don't copy old bad patterns. This code should be on form only and not bolted on to global namespaces created using `frappe.provide`